src/services/users.service.js

```diff
@@ -0,0 +1,41 @@
+const { getNumberId } = require('./../utils');
+
+class Users {
+  constructor(initialUsers) {
+    this._users = initialUsers || [];
+  }
+
+  getAll() {
+    return this._users;
+  }
+
+  getById(id) {
+    return this._users.find((user) => user.id === id) || null;
+  }
+
+  create(data) {
+    const newUser = { ...data, id: getNumberId() };
+
+    this._users.push(newUser);
+
+    return newUser;
+  }
+
+  update(id, data) {
+    const userToUpdate = this.getById(id);
```

Comment:
Check if `userToUpdate` is not null before attempting to update it to prevent errors if the user with the given ID does not exist.